### PR TITLE
Un-deprecated FlutterViewController's binaryMessenger.

### PR DESCRIPTION
**Pull request by gaaclarke**
Leaving it as a valid convenience method and to help minimize a breaking change.

References: https://github.com/flutter/flutter/issues/35928

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/FlutterViewController.mm [hunk 1]
```diff
@@ -25,6 +25,9 @@
 
 NSNotificationName const FlutterSemanticsUpdateNotification = @"FlutterSemanticsUpdate";
 
+// This is left a FlutterBinaryMessenger privately for now to give people a chance to notice the
+// change.  Unfortunately unless you have Werror turned on, incompatible pointers as arguments
```
Comment (xster) [inline shell/platform/darwin/ios/framework/Source/FlutterViewController.mm]: What's with the double spaces? :)

File: shell/platform/darwin/ios/framework/Source/FlutterViewController.mm [hunk 2]
```diff
@@ -25,6 +25,9 @@
...
+// are just a warning.
 @interface FlutterViewController () <FlutterBinaryMessenger>
```
Comment (xster) [inline shell/platform/darwin/ios/framework/Source/FlutterViewController.mm]: Can we leave a bug to remove this at some point?

**Comment by xster** [comment]
Actually, looking at the constructor, how challenging would it be to have a test that passes in an ocmock engine, call binarymessenger and verify binarymessenger call on the engine?

File: shell/platform/darwin/ios/framework/Source/FlutterViewController.mm (hunk 1 above)
Comment (gaaclarke) [reply shell/platform/darwin/ios/framework/Source/FlutterViewController.mm]: Oh man, that's going to be a hard one for me not to do.  I removed it even though it kills me.

File: shell/platform/darwin/ios/framework/Source/FlutterViewController.mm (hunk 2 above)
Comment (gaaclarke) [reply shell/platform/darwin/ios/framework/Source/FlutterViewController.mm]: https://github.com/flutter/flutter/issues/35945 done

**Comment by gaaclarke** [comment]
> Actually, looking at the constructor, how challenging would it be to have a test that passes in an ocmock engine, call binarymessenger and verify binarymessenger call on the engine?

That wouldn't be hard to do.  You are talking about the constructor of the FlutterViewController, right?  I can try to add a unit test for this.  It's probably good to get one going for FlutterViewController.

**Pull request merged**